tests/PHPStan/Rules/Cast/InvalidCastRuleTest.php

```diff
@@ -82,4 +88,84 @@ public function testCastObjectToString(): void
 		]);
 	}
 
+	public function dataMixed(): array
+	{
+		$explicitOnlyErrors = [
```

Comment:
IMO it should report all casts (possible with the exception of `(bool)` cast). But I consider that out-of-scope for this PR.